src/transactions/OfferExchange.h

```diff
@@ -8,6 +8,228 @@
+//      bother selling, and "widen out" towards the edges where the prices
+//      (should they occur) would be tempting for lots of people to sell at.
+//
+//   3. At any given moment, the order _book_ (this diagram) contains all the
+//      orders that are _not_ matched. Orders recorded in the book are the
+//      unmatched _residue_ of orders submitted for trading. In other words,
+//      when someone submits a new order, the exchange matching engine will
+//      compare it to the orders in the book and _execute_ any part it can,
+//      swapping assets from the parties involved and effectively _deleting the
+//      intersection_ of the submitted order and the order book from both,
+//      writing only the _symmetric difference_ of them back into the book.
+//      Orders that match (and should be executed / symmetric-differenced)
+//      are also called "crossing orders", and sometimes this is verbed and
+//      the entire act of matching and executing is called "crossing" a pair
+//      of orders/offers.
+//
+//   4. Because the set of orders is not perfectly smooth -- there is not
+//      necessarily every possible quantity of an asset on offer at every
+//      possible price -- deleting the symmetric difference of matching orders
+//      is likely to open up a _gap_ between the _remaining_ cheapest offers (if
+//      there are any!) in either direction of the trading pair. This gap is the
+//      _spread_ in the pair. The size of the spread will vary depending on
+//      quantity and variability of prices asked by sellers: a sparse or highly
+//      variably-priced set of offers will produce a bigger spread.
+//
+// So far so good? Ok, now let's talk about the more-detailed semantics of the
+// orders we support!
+//
+//
+// Limit orders, exact orders, and price selection
+// -----------------------------------------------
+//
+// Orders/Offers (see terminology note above) submitted to the system are
+// actually _limit_ offers, meaning that a new order will match against any
+// stored offer in he book offering the seller a _higher_ price for their merch
+// as well, just not a _lower_ one. So in the diagram above, for example, if
+// someone submits an offer to sell sheep at a "cheap" 15 wheat per sheep price,
+// then the cheapest wheat at the top of the wheat-selling stack (that priced at
+// 20 wheat per sheep, i.e. paying the sheep-seller _more_) still matches the
+// new offer and sells, even though the sheep-seller was willing to sell sheep
+// for less. An offer for a higher price was already present, and the limit
+// order on the sheep sale only bounds the price below (and amount above). It
+// will take the higher price too, better for the seller!
+//
+// What price is the sale made at? The price of the offer stored in the book. In
+// fact, offers _in the book_ are not limit offers, they are _exact_ offers that
+// strictly cross _at the price written on them_, never anything else. In the
```

Comment:
"never anything else" is actually not true, cap0003 describes this more, but rounding causes bias that can cause a better price (when the standing offer wins) or worst price (when the standing offer is taken entirely), it's also possible that the order book decides to just throw away parts of the order to avoid violating threshold (to avoid making more than 1% error on the price)

Reply:
I think you mean CAP-004. Noted and added expanded section discussing.